missile.py

```diff
@@ -1,19 +1,48 @@
+import pygame
+import math
 from constants import *
 
 class Missile(pygame.sprite.Sprite):
-    def __init__(self, x, y, mouse_x, mouse_y):
+    def __init__(self, x, y, player):
         super().__init__()
+
+        #where the bullet starts, turret area of tank image
+        self.x = x + (player.width / 2)
+        self.y = y + (player.height / 2)
+
         self.image = pygame.Surface((4, 4), pygame.SRCALPHA)
-        self.rect = self.image.get_rect(center=(x, y))
+        self.rect = self.image.get_rect(center=(self.x, self.y))
         self.color = 'white'
         #speed of missile
         self.speed = 7
-        #math that makes it work
-        self.angle = math.atan2(y-mouse_y, x-mouse_x)
+
+        #sets the direction for where the bullet goes, 
+        #depends fully on the direction the tank is facing
+        if player.direction == TANK_UP:
+            self.direction_x = player.rect.x
+            self.direction_y = player.rect.y - 10
+        elif player.direction == TANK_RIGHT:
+            self.direction_x = player.rect.x +10
+            self.direction_y = player.rect.y
+        elif player.direction == TANK_DOWN:
+            self.direction_x = player.rect.x
+            self.direction_y = player.rect.y +10
+        elif player.direction == TANK_LEFT:
+            self.direction_x = player.rect.x -10
+            self.direction_y = player.rect.y
+
+        #math that makes the bullet move
+        self.angle = math.atan2(y-self.direction_y, x-self.direction_x)
         self.x_vel = math.cos(self.angle) * self.speed
         self.y_vel = math.sin(self.angle) * self.speed
         
-    #display on screen
+    # display on screen
+    def display(self, display, player):
+        self.x -= int(self.x_vel)
+        self.y -= int(self.y_vel)
+        pygame.draw.circle(display, (255, 255, 255), (self.x, self.y), 2)
```

Comment:
This display function looks the same as the update function, so I'm  unsure why we need both.

Reply:
you're right, I think this remained in place after your work with the missile sprites. Let's get rid of this